pandas/core/missing.py

```diff
@@ -89,15 +88,11 @@ def mask_missing(arr: ArrayLike, values_to_mask) -> npt.NDArray[np.bool_]:
     # GH 21977
     mask = np.zeros(arr.shape, dtype=bool)
     for x in nonna:
-        if is_numeric_v_string_like(arr, x):
```

Comment:
I'd prefer if you pre-compute a mask in case we have object dtype before entering the loop.
Then you can add another if clause to only update the values that are not na

```
new_mask[~object_na_mask] = arr[~object_na_mask] == x
```